src/components/Stepper.tsx

```diff
@@ -22,13 +22,8 @@ export function Stepper({ steps, currentStep, onChange }: StepperBarProps) {
     throw new Error("Stepper must be initialized with at least one step");
   }
 
-  const [progressBarStep, setProgressBarStep] = useState(0);
-  useEffect(() => {
-    const currentStepIndex = steps.findIndex((s: Step) => s.value === currentStep) || 0;
-    if (currentStepIndex > progressBarStep) {
-      setProgressBarStep(currentStepIndex);
-    }
-  }, [currentStep]);
+  // calc progress based on last completed step - return -1 when no steps completed
+  const progressBarStep = steps.reduce((acc, step, idx) => (step.state === "complete" ? idx : acc), -1);
```

Comment:
Question: I can't quite tell, but is this `steps.findIndex(s => s.state === "complete")`?

Reply:
It doesn't want the first index, it wants the last index of `complete`, so maybe:
```
steps.reverse().findIndex((s) => s.state === "complete")
```

Reply:
It doesn't want the first index, it wants the last index of `complete`, so maybe:
```
steps.map((s) => s.state).lastIndexOf("complete");
```
?

Reply:
yep nice suggestion - will update to use `lastIndexOf`. Much easier to comprehend the intention of the code then the `reduce` :)